src/components/Modal/Modal.js

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useRef } from 'react';
+import './Modal.css';
+
+const Modal = ({ showModal, handleModalClose, deleteItem, item }) => {
+  const toggleModalClassName = showModal ? 'display-block' : 'display-none';
+
+  const cancelRef = useRef();
+  const deleteRef = useRef();
+
+  useEffect(() => {
+    if (showModal) {
+      // when modal opens, add eventListeners and put initial focus on "No, Cancel"
+      document.addEventListener('keydown', handleKeyEvents);
+      cancelRef.current.focus();
+    } else {
+      document.removeEventListener('keydown', handleKeyEvents);
+    }
+  }, [showModal]);
+
+  const handleKeyEvents = (e) => {
+    // close modal if user hits Escape (27)
+    if (e.keyCode === 27) {
+      handleModalClose();
+    }
+
+    // this keeps keyboard focus within modal. if user hits Tab (9)
+    if (e.keyCode === 9) {
+      // ...and NOT shift while on Delete, put focus on Cancel
+      if (!e.shiftKey && document.activeElement === deleteRef.current) {
+        cancelRef.current.focus();
+        return e.preventDefault();
+      }
+
+      // ...and shift while on Cancel, put focus on Delete
+      if (e.shiftKey && document.activeElement === cancelRef.current) {
+        deleteRef.current.focus();
+        return e.preventDefault();
+      }
+    }
+  };
```

Comment:
By moving the `handleKeyEvents` function into the `useEffect` and adding a clean up function, we ensure only we are only adding the eventlistener when it's needed and removing it when the modal component unmounts. Here's is [the recommendation on how to handle 'exhaustive-deps' lint rule](https://github.com/facebook/react/issues/14920).
```suggestion
  useEffect(() => {
    const handleKeyEvents = (e) => {
      // close modal if user hits Escape (27)
      if (e.keyCode === 27) {
        handleModalClose();
      }

      // this keeps keyboard focus within modal. if user hits Tab (9)
      if (e.keyCode === 9) {
        // ...and NOT shift while on Delete, put focus on Cancel
        if (!e.shiftKey && document.activeElement === deleteRef.current) {
          cancelRef.current.focus();
          return e.preventDefault();
        }

        // ...and shift while on Cancel, put focus on Delete
        if (e.shiftKey && document.activeElement === cancelRef.current) {
          deleteRef.current.focus();
          return e.preventDefault();
        }
      }
    };
    if (showModal) {
      // when modal opens, add eventListeners and put initial focus on "No, Cancel"
      document.addEventListener('keydown', handleKeyEvents);
      cancelRef.current.focus();
    }
    return () => document.removeEventListener('keydown', handleKeyEvents);
  }, [handleModalClose, showModal]);
```

Reply:
☝ This is a much simpler solution than the one I worked out.

Reply:
Committed your suggestion Yenly. Thank you all for your input and help in solving this, including @anderswift ! I'm definitely learning a lot about React hooks and the Effect hook through TCL. Thank you! <3